apis/core/v1alpha1/function_types.go

```diff
@@ -24,97 +24,94 @@ import (
-	Verbose *bool `json:"verbose,omitempty"`
-	// The image used where the git-init binary is (default: "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init:v0.17.3")
-	GitInitImage *string `json:"gitInitImage,omitempty"`
+	// Credentials references a Secret that contains credentials to access
+	// the repository.
+	//
+	// +optional
+	Credentials *v1.LocalObjectReference `json:"credentials,omitempty"`
 }
 
 func (gr *GitRepo) Init() {
-	var revision, refspec, subDir, sourceSubPath, httpProxy, httpsProxy, noProxy, gitInitImage string
-	var deletingExisting, submodules, sslVerify, verbose bool
-	var depth int8
+	var revision, sourceSubPath string
 	gr.Revision = &revision
-	gr.Refspec = &refspec
-	gr.SubDirectory = &subDir
 	gr.SourceSubPath = &sourceSubPath
-	gr.DeleteExisting = &deletingExisting
-	gr.HttpProxy = &httpProxy
-	gr.HttpsProxy = &httpsProxy
-	gr.NoProxy = &noProxy
-	gr.GitInitImage = &gitInitImage
-	gr.Submodules = &submodules
-	gr.SslVerify = &sslVerify
-	gr.Verbose = &verbose
-	gr.Depth = &depth
-}
-
-type Registry struct {
-	// Image registry url
-	Url *string `json:"url,omitempty"`
-	// Image registry account including username and password
-	Account *v1.SecretKeySelector `json:"account,omitempty"`
-}
-
-func (r *Registry) Init() {
-	var url string
-	r.Url = &url
-	r.Account = &v1.SecretKeySelector{}
+	gr.Credentials = &v1.LocalObjectReference{}
 }
 
 type Language string
 type Runtime string
 
 const (
 	Go            Language = "go"
-	Python        Language = "python"
 	Node          Language = "node"
 	BuildPhase             = "Build"
 	ServingPhase           = "Serving"
 	Created                = "Created"
-	Launching              = "Launching"
-	Launched               = "Launched"
+	Building               = "building"
+	Running                = "Running"
+	Succeeded              = "Succeeded"
 	Failed                 = "Failed"
+	Skipped                = "Skipped"
 	Knative       Runtime  = "Knative"
 	OpenFuncAsync Runtime  = "OpenFuncAsync"
+	Shipwright             = "Shipwright"
 )
 
+type Strategy struct {
+	// Name of the referent; More info: http://kubernetes.io/docs/user-guide/identifiers#names
+	Name string `json:"name"`
+
+	// BuildStrategyKind indicates the kind of the buildstrategy, BuildStrategy or ClusterBuildStrategy, default is BuildStrategy.
+	Kind *string `json:"kind,omitempty"`
+}
+
+type ShipwrightEngine struct {
+	// Strategy references the BuildStrategy to use to build the image.
+	// +optional
+	Strategy *Strategy `json:"strategy,omitempty"`
+	// Timeout defines the maximum amount of time the Build should take to execute.
+	//
+	// +optional
+	// +kubebuilder:validation:Format=duration
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+}
+
 type BuildImpl struct {
-	// Cloud Native Buildpacks builders
-	Builder string `json:"builder"`
-	// Environment params to pass to the builder
+	// Builder refers to the image containing the build tools inside which
```

Comment:
=>
// Builder refers to the image containing the build tools to build the source code.